src/main/scala/com/mwz/sonar/scala/scapegoat/ScapegoatSensor.scala

```diff
@@ -0,0 +1,172 @@
+        processScapegoatWarnings(context, scapegoatWarnings)
+      case Failure(ex) =>
+        log.error(
+          "Aborting the scapegoat sensor execution, " +
+          s"cause: an error occurred while reading the scapegoat report file: '$reportPath', " +
+          s"the error was: ${ex.getMessage}."
+        )
+    }
+  }
+
+  /** Returns the path to the scapegoat report for this module */
+  private[this] def getScapegoatReportPath(settings: Configuration): Path = {
+    val scalaVersion = Scala.getScalaVersion(settings)
+    val defaultScapegoatReportPath = getDefaultScapegoatReportPath(scalaVersion)
+
+    if (!settings.hasKey(ScapegoatReportPathPropertyKey)) {
+      log.info(
+        s"Missing the property: '$ScapegoatReportPathPropertyKey', " +
+        s"using the default value: '$defaultScapegoatReportPath'."
+      )
+    }
+
+    settings
+      .get(ScapegoatReportPathPropertyKey)
+      .toOption
+      .map(path => Paths.get(path))
+      .getOrElse(defaultScapegoatReportPath)
+  }
+
+  /** Process all scapegoat warnings */
+  private[this] def processScapegoatWarnings(
+    context: SensorContext,
+    scapegoatWarnings: Map[String, Seq[ScapegoatWarning]]
+  ): Unit = {
+    val activeRules = context.activeRules
+    val filesystem = context.fileSystem
+
+    scapegoatWarnings foreach { tuple =>
+      val (filename, warnings) = tuple
+      log.info(s"Saving the scapegoat warnings of the file: '$filename'.")
+
+      getModuleFile(filename, filesystem) match {
+        case Some(file) =>
+          warnings foreach { warning =>
+            log.debug(s"Saving the warning: $warning.")
+
+            // try to retrieve the SonarQube rule for this warning
+            Option(
+              activeRules.findByInternalKey(
+                ScapegoatRulesRepository.RepositoryKey,
+                warning.inspectionId
+              )
+            ) match {
+              case Some(rule) =>
+                // if the rule was found, create a new issue
+                val issue = context.newIssue().forRule(rule.ruleKey)
+
+                issue.at(
+                  issue
+                    .newLocation()
+                    .on(file)
+                    .at(file.selectLine(warning.line))
+                    .message(warning.message)
+                )
+
+                issue.save()
+              case None =>
+                // if the rule was not found, log a warning
+                if (AllScapegoatInspections.exists(inspection => inspection.id === warning.inspectionId))
+                  log.warn(
+                    s"The warning: ${warning.inspectionId}, " +
+                    "was not activated in the current quality profile."
+                  )
+                else
+                  log.warn(
+                    s"The warning: ${warning.inspectionId}, " +
+                    "does not exist in the scapegoat rules repository."
+                  )
```

Comment:
I'm somewhat interested in knowing what do you think about this two logs, especially in their seventies.

Reply:
Thanks for bringing that up - it looked good to me at a first, but then I realised that if the rules are deactivated in your quality profiles, we probably don't want to warn users about that because they most likely deactivated the rules on purpose - I'd probably change the level to debug in this case. The second one is spot on 👍

Reply:
Ok